sfepy/homogenization/recovery.py

```diff
@@ -491,233 +491,259 @@ def destroy_pool():
-                                 verbose=False, define_args=define_args)
-    if output_dir is not None:
-        conf.options.output_dir = output_dir
 
-    recovery_hook = conf.options.get('recovery_hook', None)
-    pb, corrs = None, None
-    if recovery_hook is not None:
-        recovery_hook = conf.get_function(recovery_hook)
+def get_recovery_points(region, eps0):
+    rcoors = region.domain.mesh.coors[region.get_entities(0), :]
+    rcmin = nm.min(rcoors, axis=0)
+    rcmax = nm.max(rcoors, axis=0)
+    nn = nm.round((rcmax - rcmin) / eps0)
+    if nm.prod(nn) == 0:
+        raise ValueError(
+            'incompatible recovery region and microstructure size!')
 
-        # Coefficients and correctors
-        coefs_filename = conf.options.get('coefs_filename', 'coefs')
-        coefs_filename = op.join(conf.options.get('output_dir', '.'),
-                                 coefs_filename + '.h5')
-        coefs = Coefficients.from_file_hdf5(coefs_filename)
-        corrs = get_correctors_from_file_hdf5(dump_names=coefs.save_names)
+    cs = []
+    for ii, n in enumerate(nn):
+        cs.append(nm.arange(n) * eps0 + rcmin[ii])
 
-        pb = Problem.from_conf(conf, init_equations=False, init_solvers=False)
+    ccoors = nm.empty((int(nm.prod(nn)), nn.shape[0]), dtype=nm.float64)
+    for ii, icoor in enumerate(nm.meshgrid(*cs, indexing='ij')):
+        ccoors[:, ii] = icoor.flatten()
 
-    _recovery_global_dict['micro_problem'] = pb, corrs, recovery_hook
+    return ccoors
 
 
-def recover_micro_hook(micro_filename, region, macro,
-                       naming_scheme='step_iel',
-                       recovery_file_tag='',
+def recover_micro_hook(micro_filename, region, macro, eps0,
+                       region_mode='el_centers', eval_mode='constant',
+                       eval_vars=None, corrs=None, recovery_file_tag='',
                        define_args=None, output_dir=None, verbose=False):
+    """
+    Parameters
+    ----------
+    micro_filename : str
+        The definition file of the microproblem.
+    region : Region or array
+        The macroscopic region to be recovered. If array, the centers of
+        microscopic RVEs (Representative Volume Element) are assumed to be
+        stored in it. If Region, the RVE centers are computed according to
+        `region_mode`, see below.
+    macro : dict of arrays or tuples
+        Either macroscopic values (if array) or the tuple
+        (mode, eval_var, nodal_values) is expected. The tuple is used to
+        evaluate the macroscopic values in given points of RVEs (see
+        'eval_mode`). `mode` can be 'val', 'grad', 'div', or 'cauchy_strain'.
+    eps0 : float
+        The size of the microstructures (RVE).
+    region_mode : {'el_centers', 'tiled'}
+        If 'el_centers', the RVE centers are idential to the centers of the
+        macroscopic FE mesh. If 'tiled', the recovered region is tiled
+        by rescaled RVEs.
```

Comment:
I do not really understand what this means, but OK as long as you understand it :)